streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceivedBlockTracker.scala

```diff
@@ -235,8 +350,67 @@ private[streaming] class ReceivedBlockTracker(
     }
   }
 
+  /**
+   * Creates a WAL Writer in a separate thread to enable batching of log events.
+   * Exposed for tests.
+   */
+  protected def createBatchWriteAheadLogWriter(): Option[BatchLogWriter] = {
+    if (!WriteAheadLogUtils.isBatchingEnabled(conf)) return None
+    val writer = checkpointDirOption.map(_ => new BatchLogWriter)
+    writer.foreach { runnable =>
+      new Thread(runnable, "Batch WAL Writer").start()
+      Runtime.getRuntime.addShutdownHook(new Thread {
```

Comment:
Could you remove this shutdown hook? If you find a case that`ReceivedBlockTracker` won't be stopped, it's better to fix it rather than adding a shutdown hook.